src/Components/Components/src/NavigationManager.cs

```diff
@@ -90,10 +90,47 @@ protected set
         /// <param name="uri">The destination URI. This can be absolute, or relative to the base URI
         /// (as returned by <see cref="BaseUri"/>).</param>
         /// <param name="forceLoad">If true, bypasses client-side routing and forces the browser to load the new page from the server, whether or not the URI would normally be handled by the client-side router.</param>
-        public void NavigateTo(string uri, bool forceLoad = false)
+        // 5.0 BACKCOMPAT OVERLOAD -- DO NOT TOUCH
+        public void NavigateTo(string uri, bool forceLoad)
         {
+            NavigateTo(uri, forceLoad, false);
+        }
+
+        /// <summary>
+        /// Navigates to the specified URI.
+        /// </summary>
+        /// <param name="uri">The destination URI. This can be absolute, or relative to the base URI
+        /// (as returned by <see cref="BaseUri"/>).</param>
+        /// <param name="forceLoad">If true, bypasses client-side routing and forces the browser to load the new page from the server, whether or not the URI would normally be handled by the client-side router.</param>
+        /// <param name="replace">If true, will replace the uri in the current browser history state, instead of pushing the new uri onto the browser history stack.</param>
+        public void NavigateTo(string uri, bool forceLoad = false, bool replace = false)
```

Comment:
Do we want this extra method as a convenience?, or force people who want to set the Replace flag to use the function
```cs
NavigateTo(string uri, NavigationOptions options)
``` 
Adding this function might introduce another backward compatibility problem later if we want to add even more options